app/ydoc-server/src/languageServerSession.ts

```diff
@@ -483,6 +484,14 @@ class ModulePersistence extends ObservableV2<{ removed: () => void }> {
     }
   }
 
+  private static encodeCodeSnapshot(code: string): string {
+    return Base64.encode(code)
```

Comment:
There is a nodejs [zlib](https://nodejs.org/docs/latest/api/zlib.html) module. But it will require adding zlib support for the GraalVM version. I'll implement it as a separate PR

Reply:
Filed as #11420